### PR TITLE
[MM-39640] Migrate from gorp to sqlx in store/sqlstore/role_store.go

### PR DESCRIPTION
**Pull request by shadowshot-x**
#### Summary

Migrating from gorp to sqlx in role_store.go.

#### Ticket Link

Fixes https://github.com/mattermost/mattermost-server/issues/18878

#### Release Note

```release-note
NONE
```

### CODE REVIEW

File: store/sqlstore/role_store.go [hunk 1]
```diff
@@ -167,7 +180,7 @@ func (s *SqlRoleStore) Get(roleId string) (*model.Role, error) {
 func (s *SqlRoleStore) GetAll() ([]*model.Role, error) {
 	var dbRoles []Role
```
Comment (isacikgoz) [inline store/sqlstore/role_store.go:181]: This applies to similar occurrences;
```suggestion
	dbRoles := []Role{}
```

**Review by isacikgoz — COMMENTED** [review]
Thanks for the contribution @shadowshot-x , I think we better initialize the slices upfront.

File: store/sqlstore/role_store.go (hunk 1 above)
Comment (shadowshot-x) [reply store/sqlstore/role_store.go:181]: Right @isacikgoz. Did not notice that. Correcting it.

File: store/sqlstore/role_store.go [hunk 2]
```diff
@@ -204,15 +217,15 @@ func (s *SqlRoleStore) GetByNames(names []string) ([]*model.Role, error) {
...
 	if err != nil {
 		return nil, errors.Wrap(err, "failed to find Roles")
 	}
 
-	var roles []*model.Role
+	roles := []*model.Role{}
 	defer rows.Close()
 	for rows.Next() {
-		var role Role
+		role := Role{}
```
Comment (agnivade) [inline store/sqlstore/role_store.go:228]: This is okay to keep as-is.

File: store/sqlstore/role_store.go [hunk 3]
```diff
@@ -231,8 +244,8 @@ func (s *SqlRoleStore) GetByNames(names []string) ([]*model.Role, error) {
 
 func (s *SqlRoleStore) Delete(roleId string) (*model.Role, error) {
 	// Get the role.
-	var role *Role
-	if err := s.GetReplica().SelectOne(&role, "SELECT * from Roles WHERE Id = :Id", map[string]interface{}{"Id": roleId}); err != nil {
+	role := Role{}
```
Comment (agnivade) [inline store/sqlstore/role_store.go:247]: `var role Role` is fine.

**Review by agnivade — COMMENTED** [review]
Looking great @shadowshot-x. Just some minor adjustments.

File: store/sqlstore/role_store.go (hunk 2 above)
Comment (shadowshot-x) [reply store/sqlstore/role_store.go:228]: Sure @agnivade. Doing this.

**Review by agnivade — APPROVED** [review]
Nice work @shadowshot-x !

**Comment by agnivade** [comment]
Thanks @shadowshot-x - this looks good from my side. I was waiting for Ibrahim's approval, but he has been buried in other stuff. We can merge this. Sorry for the delay!

**Pull request merged**